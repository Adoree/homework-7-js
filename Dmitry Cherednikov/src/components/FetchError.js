import React from 'react';
import PropTypes from 'prop-types';

const FetchError = ({
  message,
  onRetry
}) => (
  <div>
    <p>
      {message}
    </p>
    <button
      className="button button-load"
      onClick={onRetry}
    >
      retry
    </button>
  </div>
);

FetchError.propTypes = {
  message: PropTypes.string.isRequired,
  onRetry: PropTypes.func.isRequired
};

export default FetchError;