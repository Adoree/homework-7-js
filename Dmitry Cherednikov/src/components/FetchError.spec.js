import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import FetchError from './FetchError';

const setup = () => {
  const props = {
    message: 'Something went wrong',
    onRetry: jest.fn(),
  };

  const output = shallow((
    <FetchError { ...props } />
  ));

  return {
    output,
    props,
  }
};

describe('<FetchError />', () => {
  it('renders component', () => {
    const { output } = setup();

    expect(toJson(output)).toMatchSnapshot();
  });

  it('successfully calls the onClick handler', () => {
    const { output, props } = setup();

    output.find('button').simulate('click');
    expect(props.onRetry.mock.calls.length).toEqual(1);
  })
});
