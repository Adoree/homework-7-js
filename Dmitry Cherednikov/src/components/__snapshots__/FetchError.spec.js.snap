// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FetchError /> renders component 1`] = `
<div>
  <p>
    Something went wrong
  </p>
  <button
    className="button button-load"
    onClick={[Function]}
  >
    retry
  </button>
</div>
`;
